frontend/src/Editor/Box.jsx

```diff
@@ -102,12 +106,24 @@ export const Box = function Box({
   }
 
   const ComponentToRender = AllComponents[component.component];
-  const resolvedProperties = resolveProperties(component, currentState);
-  const resolvedStyles = resolveStyles(component, currentState);
-
+  const resolvedProperties = resolveProperties(component, currentState, null, customResolvables);
+  const resolvedStyles = resolveStyles(component, currentState, null, customResolvables);
   resolvedStyles.visibility = resolvedStyles.visibility !== false ? true : false;
 
-  const exposedVariables = currentState?.components[component.name] ?? {};
+  let exposedVariables = {};
+
+  if (component.parent) {
+    const parentComponent = components[component.parent];
```

Comment:
The variable `components` is not defined here. And I think the reason why its not causing an error is because the flow of execution is not reaching this area of the code, because, `component.parent` is always undefined on line 115. I'm not able to figure out why.